mantidproject/mantid
# Add Cropping by Fraction to ISIS Powder

**Description of work.**
When cropping a range of data, users can now crop by using fractional values as well as absolute ones to decide where the new limits of the data should be. 
Passing a tuple or list of tuples into supporting algorithms will allow for cropping. A value must be less than or equal to one for fractional cropping.

**To test:**


1. Run this script:
```python
from isis_powder import hrpd

focused_cropping_values = [
    (0.10, 0.90),           # Bank 1, Using fractional limits. 10% removed from each side
    (0.981e5, 2.0861e5),    # Bank 2
    (0.981e5, 2.0118e5)     # Bank 3
]

config_file_path = r"/mask-all-pulses/hrpd_config_example.yaml"
hrpd = hrpd.HRPD(config_file=config_file_path,user_name="test")
hrpd.create_vanadium(first_cycle_run_no="76796", window="30-130",
                     multiple_scattering=False,
                     vanadium_tof_cropping=focused_cropping_values[2],
                     suffix="_van_suffix")
hrpd.focus(run_number="77265", window="30-130",
           vanadium_normalisation = True, do_absorb_corrections=False,
           vanadium_tof_cropping=focused_cropping_values,
           focused_cropping_values=focused_cropping_values,
           suffix="_suffix")
```
2. Check results are cropped as expected. 

The following [mask-all-pulses.zip](https://github.com/mantidproject/mantid/files/3754604/mask-all-pulses.zip) contains the required files to run the example code. Some paths may need changing in the hrpd_config_example files once unzipped.

Resolves #27054 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
